src/client/Client.cc

```diff
@@ -2078,6 +2078,9 @@ void Client::handle_client_session(MClientSession *m)
 
   case CEPH_SESSION_RECALL_STATE:
     trim_caps(session, m->get_max_caps());
+    session->con->send_message(new MClientSession(
+				CEPH_SESSION_RECALL_STATEMSG_ACK,
+				m->get_seq()));
```

Comment:
Does client always send ACK immediately

Reply:
@ukernel Client will always send ACKs but I don't if its fast enough to be immediate as it calls trim_caps() first.